src/lavinmq/vhost.cr

```diff
@@ -250,6 +250,7 @@ module LavinMQ
           return false unless src.unbind(dst, f.routing_key, f.arguments)
           store_definition(f, dirty: true) if !loading && src.durable? && dst.durable?
         when AMQP::Frame::Queue::Declare
+          return false if f.queue_name.starts_with?("mqtt.") && f.arguments["x-queue-type"]? != "mqtt"
```

Comment:
This won't work. It's still possible to create a queue with a conflicting name of another type. I think we need to block it in `Client` (as it's done for `amq.`) and `Controller`. But we should probably make some kind of validation class and have a list of reserved prefixes.

Reply:
NVM, didn't comment on the latest code..

Reply:
fixed in https://github.com/cloudamqp/lavinmq/pull/766/commits/58b0b1643ac86aeaa3eac283af57657b6f04d7bb

Reply:
unnecessary of me to commit both, sorry for confusion ^^!